tests/test_hooks_intelligence.py

```
@@ -0,0 +1,420 @@
+    lines = result["result"].split("\n")
+    assert any("SALES" in l for l in lines)
+
+
+# ── agent_graduation ──
+
+from gradata.hooks.agent_graduation import main as graduation_main
+
+
+def test_agent_graduation_emits_event(tmp_path):
+    with patch.dict(os.environ, {"GRADATA_BRAIN_DIR": str(tmp_path)}):
+        with patch("gradata._events.emit") as mock_emit:
```

Ruff (0.15.9)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)